test/integration/consul-container/libs/service/helpers.go

```diff
@@ -91,7 +93,101 @@ func createAndRegisterStaticServerAndSidecar(node libcluster.Agent, httpPort int
 	return serverService, serverConnectProxy, nil
 }
 
-func CreateAndRegisterStaticServerAndSidecar(node libcluster.Agent, serviceOpts *ServiceOpts, containerArgs ...string) (Service, Service, error) {
+// createAndRegisterCustomServerAndSidecar creates a custom service from the given testcontainers.ContainerRequest
+// and a sidecar proxy for the service. The customContainerCfg parameter is used to mutate the
+// testcontainers.ContainerRequest for the sidecar proxy.
+func createAndRegisterCustomServerAndSidecar(node libcluster.Agent,
```

Comment:
I think it is more accurate to use `Service` instead of `Server`:

```suggestion
func createAndRegisterCustomServiceAndSidecar(node libcluster.Agent,
```